equinox/nn/_embedding.py

```diff
@@ -233,19 +254,22 @@ def __call__(
                 )
                 internal_rope_embedding_cache[embedding_size] = freqs_cis
 
+        freqs_cis = jax.lax.dynamic_slice_in_dim(freqs_cis, offset, seq_len)
```

Comment:
This looks wrong to me. If we hit the `else` branch of `if embedding_size in internal_rope_embedding_cache ... else:` above, then we'll compute an array of length `seq_len`, which will not all be valid when sliced into here -- we'll be indexing off the end.

Reply:
Yes, this was a bug. The fix is to ensure that `freqs_cis` is at least as long as `seq_len + offset`.